aptos-move/aptos-vm/src/verifier/transaction_arg_validation.rs

```diff
@@ -27,7 +27,8 @@ use std::{
 type ValidateArg = fn(&[u8]) -> Result<(), VMStatus>;
 
 static ALLOWED_STRUCTS: Lazy<BTreeMap<String, Option<ValidateArg>>> = Lazy::new(|| {
-    [("0x1::string::String", Some(check_string as ValidateArg))]
+    [("0x1::string::String", Some(check_string as ValidateArg)),
+        ("0x1::object::Object", None)]
```

Comment:
As discussed offline, let's add some validation here to make sure the reference is valid (using exists_at native function - https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/src/natives/object.rs#L34 or something along the line)